src/cascadia/TerminalConnection/CTerminalHandoff.cpp

```diff
@@ -0,0 +1,86 @@
+#include "pch.h"
+
+#include "CTerminalHandoff.h"
+
+// The callback function when a connection is received
+static NewHandoff _pfnHandoff = nullptr;
+// The registration ID of the class object for clean up later
+static DWORD g_cTerminalHandoffRegistration = 0;
+
+// Routine Description:
+// - Starts listening for TerminalHandoff requests by registering
+//   our class and interface with COM.
+// Arguments:
+// - pfnHandoff - Function to callback when a handoff is received
+// Return Value:
+// - S_OK, E_NOT_VALID_STATE (start called when already started) or relevant COM registration error.
+HRESULT CTerminalHandoff::s_StartListening(NewHandoff pfnHandoff) noexcept
+try
+{
+    RETURN_HR_IF(E_NOT_VALID_STATE, _pfnHandoff != nullptr);
+
+    auto classFactory = Make<SimpleClassFactory<CTerminalHandoff>>();

```

Comment:
const? (I'm like 20% confident we can const this hahaha)

Reply:
I think it is fine. We don't change anything about this pointer here. The "business one" is the one that comes out of the `.As` anyway.